From 0561aca411f7895a630ac2e72f7c8ee81549d4cc Mon Sep 17 00:00:00 2001
From: Shirayuki39 <lorddemecrius83@proton.me>
Date: Mon, 8 Dec 2025 21:41:47 +0000
Subject: [PATCH] [SQUASH]: Drop PIHooks [1/2]

* We have fenrir, which PropImitationHooks clashes with

Revert "PropImitationHooks: Block key attestation for play integrity"

This reverts commit 37940038ba2d4b3b7227d0e5a0adf53e17d2f50a.

Change-Id: Ia006dcd83dba889cf0953f2f1834182d350519dc

Revert "PropImitationHooks: Skip play Integrity props in isolated processes"

This reverts commit 71bb4e082a4f791d11cfdf86a68373363547aaf5.

Revert "PropImitationHooks: Spoof certified props to finsky as well"

This reverts commit fa0e1eed64105504d04d36fa35cd18c3ef06808c.

Revert "core: Overlay the PIF JSON update URL"

This reverts commit 209c910dc727724634db4b32ccf655891486c874.

Change-Id: I7793cc59f197d26d9390c9df6dd9bbbb409402a3

Revert "core: Make PIF user configurable [1/2]"

This reverts commit 4ddaf8dc94c1179bc36ee14b8ba522c9f56e537e.

Revert "PIHooks: Dynamically spoof props for GMS"

This reverts commit e3f5efd7ae997d0bcfd9f1d20ecb7123f6ce2dc1.

Revert "KeyProviderManager: Accept N >= 1 certs per key [1/2]"

This reverts commit c8e958964d238e95de44f91ce641017214f9adb1.

Revert "base: Add key generation for keybox spoofing [1/2]"

This reverts commit b494b22bee7dd7d45b313590d15df71ccb202c74.

Revert "Initial implementation of KeyboxImitationHooks"

This reverts commit 24c6b1bd4f9a2551be3dc3aa0eb4e90b8bb1c0f3.

Revert "PIHooks: Spoof Google Photos"

This reverts commit 7419183f19df439deb11cad5fb9b3320fc0547e5.

Revert "PropImitationHooks: Make it a hidden API"

This reverts commit 6991760f3c7fe3bc4be4ecb30eab903f8e3703ed.

Revert "PropImitationHooks: Allow user to disable prop imitation"

This reverts commit c5f426a418e8854ab28df9707652ade191c111a0.

Revert "PIHooks: Get resources from context"

This reverts commit 3b07d952c4b03a9d423a0845070c72f25dad122d.

Revert "PIHooks: Allow setting any field in certified build props"

This reverts commit 15b08372dbecca8ef8af6b87ae2a4dc900dde62a.

Revert "Introduce PropImitationHooks"

This reverts commit 942436d0e83bfe41044da2e2cd5d3c32d808bb14.

Change-Id: I3ae942b91940bab488ddd8ae55e958c00ad633ae
Signed-off-by: Shirayuki39 <lorddemecrius83@proton.me>
---
 .../app/ApplicationPackageManager.java        |   4 +-
 core/java/android/app/Instrumentation.java    |   3 -
 core/java/android/provider/Settings.java      |  21 -
 .../internal/util/PropImitationHooks.java     | 356 --------------
 .../internal/util/custom/CustomUtils.java     |  27 --
 .../internal/util/custom/IKeyboxProvider.java |  56 ---
 .../util/custom/KeyProviderManager.java       | 207 ---------
 .../util/custom/KeyboxChainGenerator.java     | 437 ------------------
 .../util/custom/KeyboxImitationHooks.java     | 177 -------
 .../internal/util/custom/KeyboxUtils.java     | 154 ------
 core/res/res/values/custom_config.xml         |  39 --
 core/res/res/values/custom_symbols.xml        |   9 -
 keystore/java/android/security/KeyStore2.java |   7 -
 .../security/KeyStoreSecurityLevel.java       |  11 -
 .../keystore2/AndroidKeyStoreSpi.java         |   3 -
 .../validators/SecureSettingsValidators.java  |   3 -
 .../server/custom/AttestationService.java     | 149 ------
 .../server/wm/ActivityTaskManagerService.java |   9 +-
 .../java/com/android/server/SystemServer.java |   6 -
 19 files changed, 3 insertions(+), 1675 deletions(-)
 delete mode 100644 core/java/com/android/internal/util/PropImitationHooks.java
 delete mode 100644 core/java/com/android/internal/util/custom/CustomUtils.java
 delete mode 100644 core/java/com/android/internal/util/custom/IKeyboxProvider.java
 delete mode 100644 core/java/com/android/internal/util/custom/KeyProviderManager.java
 delete mode 100644 core/java/com/android/internal/util/custom/KeyboxChainGenerator.java
 delete mode 100644 core/java/com/android/internal/util/custom/KeyboxImitationHooks.java
 delete mode 100644 core/java/com/android/internal/util/custom/KeyboxUtils.java
 delete mode 100644 services/core/java/com/android/server/custom/AttestationService.java

diff --git a/core/java/android/app/ApplicationPackageManager.java b/core/java/android/app/ApplicationPackageManager.java
index e22de7ba97d9..4a6a23bd6f4e 100644
--- a/core/java/android/app/ApplicationPackageManager.java
+++ b/core/java/android/app/ApplicationPackageManager.java
@@ -133,7 +133,6 @@ import com.android.internal.annotations.GuardedBy;
 import com.android.internal.annotations.VisibleForTesting;
 import com.android.internal.os.SomeArgs;
 import com.android.internal.pm.RoSystemFeatures;
-import com.android.internal.util.PropImitationHooks;
 import com.android.internal.util.UserIcons;
 
 import dalvik.system.VMRuntime;
@@ -844,8 +843,7 @@ public class ApplicationPackageManager extends PackageManager {
                 return maybeHasSystemFeature;
             }
         }
-        return PropImitationHooks.hasSystemFeature(name,
-                mHasSystemFeatureCache.query(new HasSystemFeatureQuery(name, version)));
+        return mHasSystemFeatureCache.query(new HasSystemFeatureQuery(name, version));
     }
 
     /** @hide */
diff --git a/core/java/android/app/Instrumentation.java b/core/java/android/app/Instrumentation.java
index c09af3834ebd..d90db96819aa 100644
--- a/core/java/android/app/Instrumentation.java
+++ b/core/java/android/app/Instrumentation.java
@@ -68,7 +68,6 @@ import android.view.Window;
 import android.view.WindowManagerGlobal;
 
 import com.android.internal.content.ReferrerIntent;
-import com.android.internal.util.PropImitationHooks;
 
 import java.io.File;
 import java.lang.annotation.Retention;
@@ -1350,7 +1349,6 @@ public class Instrumentation {
         Application app = getFactory(context.getPackageName())
                 .instantiateApplication(cl, className);
         app.attach(context);
-        PropImitationHooks.setProps(context);
         return app;
     }
     
@@ -1368,7 +1366,6 @@ public class Instrumentation {
             ClassNotFoundException {
         Application app = (Application)clazz.newInstance();
         app.attach(context);
-        PropImitationHooks.setProps(context);
         return app;
     }
 
diff --git a/core/java/android/provider/Settings.java b/core/java/android/provider/Settings.java
index af807de3bd08..efc179c7f182 100644
--- a/core/java/android/provider/Settings.java
+++ b/core/java/android/provider/Settings.java
@@ -13490,27 +13490,6 @@ public final class Settings {
          */
         public static final String BERRY_BLACK_THEME = "berry_black_theme";
 
-        /**
-         * User selectable PIF data.
-         * @hide
-         */
-        @Readable
-        public static final String PIF_DATA = "pif_data";
-
-        /**
-         * PIF data fetched from our repo.
-         * @hide
-         */
-        @Readable
-        public static final String FETCHED_PIF = "fetched_pif";
-
-        /**
-         * User selectable keybox data.
-         * @hide
-         */
-        @Readable
-        public static final String KEYBOX_DATA = "keybox_data";
-
         /**
          * Keys we no longer back up under the current schema, but want to continue to
          * process when restoring historical backup datasets.
diff --git a/core/java/com/android/internal/util/PropImitationHooks.java b/core/java/com/android/internal/util/PropImitationHooks.java
deleted file mode 100644
index 901392220410..000000000000
--- a/core/java/com/android/internal/util/PropImitationHooks.java
+++ /dev/null
@@ -1,356 +0,0 @@
-/*
- * Copyright (C) 2022-2024 Paranoid Android
- *           (C) 2023 ArrowOS
- *           (C) 2023 The LibreMobileOS Foundation
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package com.android.internal.util;
-
-import android.app.ActivityTaskManager;
-import android.app.Application;
-import android.app.TaskStackListener;
-import android.content.ComponentName;
-import android.content.Context;
-import android.content.res.Resources;
-import android.os.Build;
-import android.os.Binder;
-import android.os.Environment;
-import android.os.Process;
-import android.os.SystemProperties;
-import android.provider.Settings;
-import android.text.TextUtils;
-import android.util.Log;
-
-import com.android.internal.R;
-import com.android.internal.util.custom.KeyProviderManager;
-
-import org.json.JSONException;
-import org.json.JSONObject;
-
-import java.io.BufferedReader;
-import java.io.File;
-import java.io.FileReader;
-import java.io.IOException;
-import java.lang.reflect.Field;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Set;
-
-/**
- * @hide
- */
-public class PropImitationHooks {
-
-    private static final String TAG = "PropImitationHooks";
-    private static final boolean DEBUG = Log.isLoggable(TAG, Log.DEBUG);
-
-    private static final Boolean sDisableGmsProps = SystemProperties.getBoolean(
-            "persist.sys.pihooks.disable.gms_props", false);
-
-    private static final Boolean sDisableKeyAttestationBlock = SystemProperties.getBoolean(
-            "persist.sys.pihooks.disable.gms_key_attestation_block", false);
-    private static final String DATA_FILE = "gms_certified_props.json";
-
-    private static final String PACKAGE_ARCORE = "com.google.ar.core";
-    private static final String PACKAGE_FINSKY = "com.android.vending";
-    private static final String PACKAGE_GMS = "com.google.android.gms";
-    private static final String PROCESS_GMS_UNSTABLE = PACKAGE_GMS + ".unstable";
-    private static final String PACKAGE_NETFLIX = "com.netflix.mediaclient";
-    private static final String PACKAGE_GPHOTOS = "com.google.android.apps.photos";
-
-    private static final ComponentName GMS_ADD_ACCOUNT_ACTIVITY = ComponentName.unflattenFromString(
-            "com.google.android.gms/.auth.uiflows.minutemaid.MinuteMaidActivity");
-
-    private static final String FEATURE_NEXUS_PRELOAD =
-            "com.google.android.apps.photos.NEXUS_PRELOAD";
-
-    private static final Map<String, String> sPixelOneProps = Map.of(
-        "PRODUCT", "sailfish",
-        "DEVICE", "sailfish",
-        "MANUFACTURER", "Google",
-        "BRAND", "google",
-        "MODEL", "Pixel",
-        "FINGERPRINT", "google/sailfish/sailfish:10/QP1A.191005.007.A3/5972272:user/release-keys"
-    );
-
-    private static final Set<String> sPixelFeatures = Set.of(
-        "PIXEL_2017_EXPERIENCE",
-        "PIXEL_2017_PRELOAD",
-        "PIXEL_2018_EXPERIENCE",
-        "PIXEL_2018_PRELOAD",
-        "PIXEL_2019_EXPERIENCE",
-        "PIXEL_2019_MIDYEAR_EXPERIENCE",
-        "PIXEL_2019_MIDYEAR_PRELOAD",
-        "PIXEL_2019_PRELOAD",
-        "PIXEL_2020_EXPERIENCE",
-        "PIXEL_2020_MIDYEAR_EXPERIENCE",
-        "PIXEL_2021_MIDYEAR_EXPERIENCE"
-    );
-
-    private static final Set<String> sTensorFeatures = Set.of(
-        "PIXEL_2021_EXPERIENCE",
-        "PIXEL_2022_EXPERIENCE",
-        "PIXEL_2022_MIDYEAR_EXPERIENCE",
-        "PIXEL_2023_EXPERIENCE",
-        "PIXEL_2023_MIDYEAR_EXPERIENCE",
-        "PIXEL_2024_EXPERIENCE",
-        "PIXEL_2024_MIDYEAR_EXPERIENCE"
-    );
-
-    private static volatile List<String> sCertifiedProps = new ArrayList<>();
-    private static volatile String sStockFp, sNetflixModel;
-
-    private static volatile String sProcessName;
-    private static volatile boolean sIsGms, sIsFinsky, sIsPhotos;
-
-    public static void setProps(Context context) {
-        final String packageName = context.getPackageName();
-        final String processName = Application.getProcessName();
-
-        if (TextUtils.isEmpty(packageName) || TextUtils.isEmpty(processName)) {
-            Log.e(TAG, "Null package or process name");
-            return;
-        }
-
-        final Resources res = context.getResources();
-        if (res == null) {
-            Log.e(TAG, "Null resources");
-            return;
-        }
-
-        sStockFp = res.getString(R.string.config_stockFingerprint);
-        sNetflixModel = res.getString(R.string.config_netflixSpoofModel);
-
-        sProcessName = processName;
-        sIsGms = packageName.equals(PACKAGE_GMS) && processName.equals(PROCESS_GMS_UNSTABLE);
-        sIsFinsky = packageName.equals(PACKAGE_FINSKY);
-        sIsPhotos = packageName.equals(PACKAGE_GPHOTOS);
-
-        /* Set Certified Properties for GMSCore
-         * Set Stock Fingerprint for ARCore
-         * Set custom model for Netflix
-         * Set Pixel XL for Google Photos
-         */
-        if (sIsGms || sIsFinsky) {
-            if (!android.os.Process.isIsolated()) {
-                setPlayIntegrityProps(context);
-            } else {
-                dlog("Not setting Play Integrity props in isolated process");
-            }
-        } else if (!sStockFp.isEmpty() && packageName.equals(PACKAGE_ARCORE)) {
-            dlog("Setting stock fingerprint for: " + packageName);
-            setPropValue("FINGERPRINT", sStockFp);
-        } else if (sIsPhotos) {
-            dlog("Spoofing Pixel 1 for Google Photos");
-            sPixelOneProps.forEach((PropImitationHooks::setPropValue));
-        } else if (!sNetflixModel.isEmpty() && packageName.equals(PACKAGE_NETFLIX)) {
-            dlog("Setting model to " + sNetflixModel + " for Netflix");
-            setPropValue("MODEL", sNetflixModel);
-        }
-    }
-
-    private static void setPropValue(String key, String value) {
-        try {
-            dlog("Setting prop " + key + " to " + value.toString());
-            Class clazz = Build.class;
-            if (key.startsWith("VERSION.")) {
-                clazz = Build.VERSION.class;
-                key = key.substring(8);
-            }
-            Field field = clazz.getDeclaredField(key);
-            field.setAccessible(true);
-            // Cast the value to int if it's an integer field, otherwise string.
-            field.set(null, field.getType().equals(Integer.TYPE) ? Integer.parseInt(value) : value);
-            field.setAccessible(false);
-        } catch (Exception e) {
-            Log.e(TAG, "Failed to set prop " + key, e);
-        }
-    }
-
-    private static void setPlayIntegrityProps(Context context) {
-        if (sDisableGmsProps) {
-            dlog("GMS prop imitation is disabled by user");
-            return;
-        }
-
-        // Guard: isolated processes cannot access content providers (Settings.*).
-        if (android.os.Process.isIsolated()) {
-            dlog("Skipping setPlayIntegrityProps in isolated process");
-            return;
-        }
-
-        String savedProps = Settings.Secure.getString(context.getContentResolver(), Settings.Secure.PIF_DATA);
-        if (savedProps == null || TextUtils.isEmpty(savedProps)) {
-            savedProps = Settings.Secure.getString(context.getContentResolver(), Settings.Secure.FETCHED_PIF);
-        }
-
-        if (savedProps == null || TextUtils.isEmpty(savedProps)) {
-            dlog("Parsing props locally - fetched pif / user provided pif unavailable");
-            sCertifiedProps = Arrays.asList(context.getResources().getStringArray(R.array.config_certifiedBuildProperties));
-        } else {
-            dlog("Parsing props fetched / provided by user");
-            try {
-                JSONObject parsedProps = new JSONObject(savedProps);
-                Iterator<String> keys = parsedProps.keys();
-                while (keys.hasNext()) {
-                    String key = keys.next();
-                    String value = parsedProps.getString(key);
-                    sCertifiedProps.add(key + ":" + value);
-                }
-            } catch (JSONException e) {
-                Log.e(TAG, "Error parsing JSON data", e);
-                dlog("Parsing props locally as fallback");
-                sCertifiedProps = Arrays.asList(context.getResources().getStringArray(R.array.config_certifiedBuildProperties));
-            }
-        }
-
-        if (sCertifiedProps.isEmpty()) {
-            dlog("Certified props are not set");
-            return;
-        }
-
-        final boolean was = isGmsAddAccountActivityOnTop();
-        final TaskStackListener taskStackListener = new TaskStackListener() {
-            @Override
-            public void onTaskStackChanged() {
-                final boolean is = isGmsAddAccountActivityOnTop();
-                if (is ^ was) {
-                    dlog("GmsAddAccountActivityOnTop is:" + is + " was:" + was +
-                            ", killing myself!"); // process will restart automatically later
-                    Process.killProcess(Process.myPid());
-                }
-            }
-        };
-
-        if (!was) {
-            dlog("Spoofing build for GMS / Finsky");
-            setCertifiedProps();
-        } else {
-            dlog("Skip spoofing build for GMS / Finsky, because GmsAddAccountActivityOnTop");
-        }
-
-        try {
-            ActivityTaskManager.getService().registerTaskStackListener(taskStackListener);
-        } catch (Exception e) {
-            Log.e(TAG, "Failed to register task stack listener!", e);
-        }
-    }
-
-    private static void setCertifiedProps() {
-        for (String entry : sCertifiedProps) {
-            // Each entry must be of the format FIELD:value
-            final String[] fieldAndProp = entry.split(":", 2);
-            if (fieldAndProp.length != 2) {
-                Log.e(TAG, "Invalid entry in certified props: " + entry);
-                continue;
-            }
-            setPropValue(fieldAndProp[0], fieldAndProp[1]);
-        }
-    }
-
-    private static String readFromFile(File file) {
-        StringBuilder content = new StringBuilder();
-
-        if (file.exists()) {
-            try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
-                String line;
-
-                while ((line = reader.readLine()) != null) {
-                    content.append(line);
-                }
-            } catch (IOException e) {
-                Log.e(TAG, "Error reading from file", e);
-            }
-        }
-        return content.toString();
-    }
-
-    private static boolean isGmsAddAccountActivityOnTop() {
-        try {
-            final ActivityTaskManager.RootTaskInfo focusedTask =
-                    ActivityTaskManager.getService().getFocusedRootTaskInfo();
-            return focusedTask != null && focusedTask.topActivity != null
-                    && focusedTask.topActivity.equals(GMS_ADD_ACCOUNT_ACTIVITY);
-        } catch (Exception e) {
-            Log.e(TAG, "Unable to get top activity!", e);
-        }
-        return false;
-    }
-
-    public static boolean shouldBypassTaskPermission(Context context) {
-        if (sDisableGmsProps) {
-            return false;
-        }
-
-        // GMS doesn't have MANAGE_ACTIVITY_TASKS permission
-        final int callingUid = Binder.getCallingUid();
-        final int gmsUid;
-        try {
-            gmsUid = context.getPackageManager().getApplicationInfo(PACKAGE_GMS, 0).uid;
-            dlog("shouldBypassTaskPermission: gmsUid:" + gmsUid + " callingUid:" + callingUid);
-        } catch (Exception e) {
-            Log.e(TAG, "shouldBypassTaskPermission: unable to get gms uid", e);
-            return false;
-        }
-        return gmsUid == callingUid;
-    }
-
-    private static boolean isCallerPlayIntegrity() {
-        return Arrays.stream(Thread.currentThread().getStackTrace())
-                .map(StackTraceElement::getClassName)
-                .anyMatch(name -> name.toLowerCase(Locale.US).contains("droidguard"));
-    }
-
-    public static void onEngineGetCertificateChain() {
-        if (sDisableKeyAttestationBlock) {
-            dlog("Key attestation blocking is disabled by user");
-            return;
-        }
-
-        // If a keybox is found, don't block key attestation
-        if (KeyProviderManager.isKeyboxAvailable()) {
-            dlog("Key attestation blocking is disabled because a keybox is defined to spoof");
-            return;
-        }
-
-        // Check stack for Play Integrity
-        if (isCallerPlayIntegrity()) {
-            dlog("Blocked key attestation for play integrity");
-            throw new UnsupportedOperationException();
-        }
-    }
-
-    public static boolean hasSystemFeature(String name, boolean has) {
-        if (sIsPhotos) {
-            if (has && (sPixelFeatures.stream().anyMatch(name::contains)
-                    || sTensorFeatures.stream().anyMatch(name::contains))) {
-                dlog("Blocked system feature " + name + " for Google Photos");
-                has = false;
-            } else if (!has && name.equalsIgnoreCase(FEATURE_NEXUS_PRELOAD)) {
-                dlog("Enabled system feature " + name + " for Google Photos");
-                has = true;
-            }
-        }
-        return has;
-    }
-
-    public static void dlog(String msg) {
-        if (DEBUG) Log.d(TAG, "[" + sProcessName + "] " + msg);
-    }
-}
diff --git a/core/java/com/android/internal/util/custom/CustomUtils.java b/core/java/com/android/internal/util/custom/CustomUtils.java
deleted file mode 100644
index 5e28ebf6b87f..000000000000
--- a/core/java/com/android/internal/util/custom/CustomUtils.java
+++ /dev/null
@@ -1,27 +0,0 @@
-package com.android.internal.util.custom;
-
-import android.content.Context;
-import android.content.pm.ApplicationInfo;
-import android.content.pm.PackageInfo;
-import android.content.pm.PackageManager;
-
-public class CustomUtils {
-
-    public static boolean isPackageInstalled(Context context, String packageName, boolean ignoreState) {
-        if (packageName != null) {
-            try {
-                PackageInfo pi = context.getPackageManager().getPackageInfo(packageName, 0);
-                if (!pi.applicationInfo.enabled && !ignoreState) {
-                    return false;
-                }
-            } catch (PackageManager.NameNotFoundException e) {
-                return false;
-            }
-        }
-        return true;
-    }
-
-    public static boolean isPackageInstalled(Context context, String packageName) {
-        return isPackageInstalled(context, packageName, true);
-    }
-}
diff --git a/core/java/com/android/internal/util/custom/IKeyboxProvider.java b/core/java/com/android/internal/util/custom/IKeyboxProvider.java
deleted file mode 100644
index deafa29e40c7..000000000000
--- a/core/java/com/android/internal/util/custom/IKeyboxProvider.java
+++ /dev/null
@@ -1,56 +0,0 @@
-/*
- * SPDX-FileCopyrightText: 2024 Paranoid Android
- * SPDX-License-Identifier: Apache-2.0
- */
-package com.android.internal.util.custom;
-
-/**
- * Interface for keybox providers.
- *
- * This interface defines the methods that a keybox provider must implement
- * to provide access to EC and RSA keys and certificate chains.
- *
- * @hide
- */
-public interface IKeyboxProvider {
-
-    /**
-     * Checks if a valid keybox is available.
-     *
-     * @return true if a valid keybox is available, false otherwise
-     * @hide
-     */
-    boolean hasKeybox();
-
-    /**
-     * Retrieves the EC private key.
-     *
-     * @return the EC private key as a String
-     * @hide
-     */
-    String getEcPrivateKey();
-
-    /**
-     * Retrieves the RSA private key.
-     *
-     * @return the RSA private key as a String
-     * @hide
-     */
-    String getRsaPrivateKey();
-
-    /**
-     * Retrieves the EC certificate chain.
-     *
-     * @return an array of Strings representing the EC certificate chain
-     * @hide
-     */
-    String[] getEcCertificateChain();
-
-    /**
-     * Retrieves the RSA certificate chain.
-     *
-     * @return an array of Strings representing the RSA certificate chain
-     * @hide
-     */
-    String[] getRsaCertificateChain();
-}
diff --git a/core/java/com/android/internal/util/custom/KeyProviderManager.java b/core/java/com/android/internal/util/custom/KeyProviderManager.java
deleted file mode 100644
index 35364b452c29..000000000000
--- a/core/java/com/android/internal/util/custom/KeyProviderManager.java
+++ /dev/null
@@ -1,207 +0,0 @@
-/*
- * SPDX-FileCopyrightText: 2024 Paranoid Android
- * SPDX-License-Identifier: Apache-2.0
- */
-package com.android.internal.util.custom;
-
-import android.app.ActivityThread;
-import android.content.Context;
-import android.provider.Settings;
-import android.util.Log;
-import android.util.Xml;
-
-import com.android.internal.R;
-import org.xmlpull.v1.XmlPullParser;
-
-import java.io.StringReader;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-
-/**
- * Manager class for handling keybox providers.
- * @hide
- */
-public final class KeyProviderManager {
-    private static final String TAG = "KeyProviderManager";
-
-    private KeyProviderManager() {}
-
-    public static IKeyboxProvider getProvider() {
-        return new DefaultKeyboxProvider();
-    }
-
-    public static boolean isKeyboxAvailable() {
-        return getProvider().hasKeybox();
-    }
-
-    private static class DefaultKeyboxProvider implements IKeyboxProvider {
-        private final Map<String, String> keyboxData = new HashMap<>();
-
-        private DefaultKeyboxProvider() {
-            Context context = getApplicationContext();
-            if (context == null) {
-                Log.e(TAG, "Failed to get application context");
-                return;
-            }
-
-            if (!loadFromXmlSetting(context)) {
-                loadFromConfigArray(context);
-            }
-        }
-
-        private boolean loadFromXmlSetting(Context ctx) {
-            try {
-                String xml = Settings.Secure.getString(ctx.getContentResolver(), Settings.Secure.KEYBOX_DATA);
-                if (xml == null || xml.trim().isEmpty()) return false;
-
-                XmlPullParser p = Xml.newPullParser();
-                p.setInput(new StringReader(xml));
-
-                String currentAlg = null;
-                int certCount = 0;
-                boolean numberOfKeyboxesChecked = false;
-
-                for (int ev = p.next(); ev != XmlPullParser.END_DOCUMENT; ev = p.next()) {
-                    if (ev == XmlPullParser.START_TAG) {
-                        String tag = p.getName();
-                        switch (tag) {
-                            case "NumberOfKeyboxes":
-                                p.next();
-                                numberOfKeyboxesChecked = true;
-                                try {
-                                    int count = Integer.parseInt(p.getText().trim());
-                                    if (count != 1) {
-                                        Log.w(TAG, "Invalid NumberOfKeyboxes: " + count);
-                                        return false;
-                                    }
-                                } catch (NumberFormatException e) {
-                                    Log.w(TAG, "Failed to parse NumberOfKeyboxes", e);
-                                    return false;
-                                }
-                                break;
-
-                            case "Key":
-                                currentAlg = p.getAttributeValue(null, "algorithm");
-                                if ("ecdsa".equalsIgnoreCase(currentAlg)) currentAlg = "EC";
-                                else if ("rsa".equalsIgnoreCase(currentAlg)) currentAlg = "RSA";
-                                else currentAlg = null;
-                                certCount = 0;
-                                break;
-
-                            case "PrivateKey": {
-                                String format = p.getAttributeValue(null, "format");
-                                if (!"pem".equalsIgnoreCase(format)) {
-                                    Log.w(TAG, "Unsupported PrivateKey format: " + format);
-                                    return false;
-                                }
-                                p.next();
-                                if (currentAlg != null) {
-                                    keyboxData.put(currentAlg + ".PRIV", p.getText().trim());
-                                }
-                                break;
-                            }
-
-                            case "Certificate": {
-                                String format = p.getAttributeValue(null, "format");
-                                if (!"pem".equalsIgnoreCase(format)) {
-                                    Log.w(TAG, "Unsupported Certificate format: " + format);
-                                    return false;
-                                }
-                                if (currentAlg != null) {
-                                    p.next();
-                                    certCount++;
-                                    keyboxData.put(currentAlg + ".CERT_" + certCount, p.getText().trim());
-                                }
-                                break;
-                            }
-                        }
-                    }
-                }
-
-                if (!numberOfKeyboxesChecked) {
-                    Log.w(TAG, "Missing <NumberOfKeyboxes> in keybox XML");
-                    return false;
-                }
-
-                if (!hasKeybox()) {
-                    Log.w(TAG, "Failed to load keybox from XML setting");
-                    return false;
-                }
-
-                Log.i(TAG, "Loaded keybox from XML setting");
-                return true;
-            } catch (Exception e) {
-                Log.e(TAG, "XML keybox load failed", e);
-                return false;
-            }
-        }
-
-        private void loadFromConfigArray(Context ctx) {
-            for (String entry : ctx.getResources().getStringArray(R.array.config_certifiedKeybox)) {
-                String[] parts = entry.split(":", 2);
-                if (parts.length == 2) {
-                    keyboxData.put(parts[0], parts[1]);
-                }
-            }
-
-            if (!hasKeybox()) {
-                Log.w(TAG, "Incomplete keybox provided by overlays");
-            }
-        }
-
-        private static Context getApplicationContext() {
-            try {
-                return ActivityThread.currentApplication().getApplicationContext();
-            } catch (Exception e) {
-                Log.e(TAG, "Error getting application context", e);
-                return null;
-            }
-        }
-
-        @Override
-        public boolean hasKeybox() {
-            if (!keyboxData.containsKey("EC.PRIV") || !keyboxData.containsKey("RSA.PRIV")) {
-                return false;
-            }
-            if (!keyboxData.containsKey("EC.CERT_1") || !keyboxData.containsKey("RSA.CERT_1")) {
-                return false;
-            }
-            return true;
-        }
-
-        @Override
-        public String getEcPrivateKey() {
-            return keyboxData.get("EC.PRIV");
-        }
-
-        @Override
-        public String getRsaPrivateKey() {
-            return keyboxData.get("RSA.PRIV");
-        }
-
-        @Override
-        public String[] getEcCertificateChain() {
-            return getCertificateChain("EC");
-        }
-
-        @Override
-        public String[] getRsaCertificateChain() {
-            return getCertificateChain("RSA");
-        }
-
-        private String[] getCertificateChain(String prefix) {
-            List<String> dataList = new ArrayList<>();
-            for (String key : keyboxData.keySet()) {
-                if (key.startsWith(prefix + ".CERT_")) {
-                    dataList.add(keyboxData.get(key));
-                }
-            }
-            String[] chain = dataList.toArray(String[]::new);
-            Arrays.sort(chain);
-            return chain;
-        }
-    }
-}
diff --git a/core/java/com/android/internal/util/custom/KeyboxChainGenerator.java b/core/java/com/android/internal/util/custom/KeyboxChainGenerator.java
deleted file mode 100644
index ad6e08c9ea9a..000000000000
--- a/core/java/com/android/internal/util/custom/KeyboxChainGenerator.java
+++ /dev/null
@@ -1,437 +0,0 @@
-/*
- * SPDX-FileCopyrightText: 2025 Neoteric OS
- * SPDX-License-Identifier: Apache-2.0
- */
-package com.android.internal.util.custom;
-
-import android.app.ActivityThread;
-import android.content.Context;
-import android.content.pm.PackageInfo;
-import android.content.pm.PackageManager;
-import android.content.pm.Signature;
-import android.hardware.security.keymint.Algorithm;
-import android.hardware.security.keymint.EcCurve;
-import android.hardware.security.keymint.KeyParameter;
-import android.hardware.security.keymint.Tag;
-import android.os.Binder;
-import android.os.Build;
-import android.security.keystore.KeyProperties;
-import android.system.keystore2.KeyDescriptor;
-import android.util.Log;
-
-import androidx.annotation.Nullable;
-
-import com.android.internal.org.bouncycastle.asn1.ASN1Boolean;
-import com.android.internal.org.bouncycastle.asn1.ASN1Encodable;
-import com.android.internal.org.bouncycastle.asn1.ASN1Enumerated;
-import com.android.internal.org.bouncycastle.asn1.ASN1Integer;
-import com.android.internal.org.bouncycastle.asn1.ASN1ObjectIdentifier;
-import com.android.internal.org.bouncycastle.asn1.ASN1OctetString;
-import com.android.internal.org.bouncycastle.asn1.ASN1Sequence;
-import com.android.internal.org.bouncycastle.asn1.DERNull;
-import com.android.internal.org.bouncycastle.asn1.DEROctetString;
-import com.android.internal.org.bouncycastle.asn1.DERSequence;
-import com.android.internal.org.bouncycastle.asn1.DERSet;
-import com.android.internal.org.bouncycastle.asn1.DERTaggedObject;
-import com.android.internal.org.bouncycastle.asn1.x500.X500Name;
-import com.android.internal.org.bouncycastle.asn1.x509.Extension;
-import com.android.internal.org.bouncycastle.asn1.x509.KeyUsage;
-import com.android.internal.org.bouncycastle.asn1.x509.SubjectPublicKeyInfo;
-import com.android.internal.org.bouncycastle.asn1.x509.Time;
-import com.android.internal.org.bouncycastle.cert.X509CertificateHolder;
-import com.android.internal.org.bouncycastle.cert.X509v3CertificateBuilder;
-import com.android.internal.org.bouncycastle.jce.provider.BouncyCastleProvider;
-import com.android.internal.org.bouncycastle.operator.ContentSigner;
-import com.android.internal.org.bouncycastle.operator.jcajce.JcaContentSignerBuilder;
-
-import java.io.IOException;
-import java.math.BigInteger;
-import java.nio.charset.StandardCharsets;
-import java.security.KeyPair;
-import java.security.KeyPairGenerator;
-import java.security.MessageDigest;
-import java.security.SecureRandom;
-import java.security.Security;
-import java.security.cert.Certificate;
-import java.security.spec.ECGenParameterSpec;
-import java.security.spec.RSAKeyGenParameterSpec;
-import java.util.Arrays;
-import java.util.ArrayList;
-import java.util.Date;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Objects;
-import java.util.Set;
-
-import javax.security.auth.x500.X500Principal;
-
-/**
- * @hide
- */
-public final class KeyboxChainGenerator {
-
-    private static final String TAG = "KeyboxChainGenerator";
-    private static final boolean DEBUG = Log.isLoggable(TAG, Log.DEBUG);
-
-    private static final int ATTESTATION_APPLICATION_ID_PACKAGE_INFOS_INDEX = 0;
-    private static final int ATTESTATION_APPLICATION_ID_SIGNATURE_DIGESTS_INDEX = 1;
-    private static final int ATTESTATION_PACKAGE_INFO_PACKAGE_NAME_INDEX = 0;
-    private static final int ATTESTATION_PACKAGE_INFO_VERSION_INDEX = 1;
-
-    public static List<Certificate> generateCertChain(int uid, KeyDescriptor descriptor, KeyGenParameters params) {
-        dlog("Requested KeyPair with alias: " + descriptor.alias);
-        int size = params.keySize;
-        KeyPair kp;
-        try {
-            if (Objects.equals(params.algorithm, Algorithm.EC)) {
-                dlog("Generating EC keypair of size " + size);
-                kp = buildECKeyPair(params);
-            } else if (Objects.equals(params.algorithm, Algorithm.RSA)) {
-                dlog("Generating RSA keypair of size " + size);
-                kp = buildRSAKeyPair(params);
-            } else {
-                dlog("Unsupported algorithm");
-                return null;
-            }
-
-            X509v3CertificateBuilder certBuilder = new X509v3CertificateBuilder(
-                    KeyboxUtils.getCertificateHolder(
-                            Objects.equals(params.algorithm, Algorithm.EC)
-                                    ? KeyProperties.KEY_ALGORITHM_EC
-                                    : KeyProperties.KEY_ALGORITHM_RSA
-                    ).getSubject(),
-                    params.certificateSerial,
-                    new Time(params.certificateNotBefore),
-                    new Time(params.certificateNotAfter),
-                    params.certificateSubject,
-                    SubjectPublicKeyInfo.getInstance(
-                            ASN1Sequence.getInstance(kp.getPublic().getEncoded())
-                    )
-            );
-
-            KeyUsage keyUsage = new KeyUsage(KeyUsage.keyCertSign);
-            certBuilder.addExtension(Extension.keyUsage, true, keyUsage);
-            certBuilder.addExtension(createExtension(params, uid));
-
-            ContentSigner contentSigner;
-            if (Objects.equals(params.algorithm, Algorithm.EC)) {
-                contentSigner = new JcaContentSignerBuilder("SHA256withECDSA").build(KeyboxUtils.getPrivateKey(KeyProperties.KEY_ALGORITHM_EC));
-            } else {
-                contentSigner = new JcaContentSignerBuilder("SHA256withRSA").build(KeyboxUtils.getPrivateKey(KeyProperties.KEY_ALGORITHM_RSA));
-            }
-            X509CertificateHolder certHolder = certBuilder.build(contentSigner);
-            Certificate leaf = KeyboxUtils.getCertificateFromHolder(certHolder);
-            List<Certificate> chain = KeyboxUtils.getCertificateChain(leaf.getPublicKey().getAlgorithm());
-            chain.add(0, leaf);
-            dlog("Successfully generated X500 Cert for alias: " + descriptor.alias);
-            return chain;
-        } catch (Throwable t) {
-            Log.e(TAG, Log.getStackTraceString(t));
-        }
-        return null;
-    }
-
-    private static ASN1Encodable[] fromIntList(List<Integer> list) {
-        ASN1Encodable[] result = new ASN1Encodable[list.size()];
-        for (int i = 0; i < list.size(); i++) {
-            result[i] = new ASN1Integer(list.get(i));
-        }
-        return result;
-    }
-
-    private static Extension createExtension(KeyGenParameters params, int uid) {
-        try {
-            SecureRandom random = new SecureRandom();
-
-            byte[] bytes1 = new byte[32];
-            byte[] bytes2 = new byte[32];
-
-            random.nextBytes(bytes1);
-            random.nextBytes(bytes2);
-
-            ASN1Encodable[] rootOfTrustEncodables = {new DEROctetString(bytes1), ASN1Boolean.TRUE,
-                    new ASN1Enumerated(0), new DEROctetString(bytes2)};
-
-            ASN1Sequence rootOfTrustSeq = new DERSequence(rootOfTrustEncodables);
-
-            var Apurpose = new DERSet(fromIntList(params.purpose));
-            var Aalgorithm = new ASN1Integer(params.algorithm);
-            var AkeySize = new ASN1Integer(params.keySize);
-            var Adigest = new DERSet(fromIntList(params.digest));
-            var AecCurve = new ASN1Integer(params.ecCurve);
-            var AnoAuthRequired = DERNull.INSTANCE;
-
-            // To be loaded
-            var AosVersion = new ASN1Integer(getOsVersion());
-            var AosPatchLevel = new ASN1Integer(getPatchLevel());
-
-            var AapplicationID = createApplicationId(uid);
-            var AbootPatchlevel = new ASN1Integer(getPatchLevelLong());
-            var AvendorPatchLevel = new ASN1Integer(getPatchLevelLong());
-
-            var AcreationDateTime = new ASN1Integer(System.currentTimeMillis());
-            var Aorigin = new ASN1Integer(0);
-
-            var purpose = new DERTaggedObject(true, 1, Apurpose);
-            var algorithm = new DERTaggedObject(true, 2, Aalgorithm);
-            var keySize = new DERTaggedObject(true, 3, AkeySize);
-            var digest = new DERTaggedObject(true, 5, Adigest);
-            var ecCurve = new DERTaggedObject(true, 10, AecCurve);
-            var noAuthRequired = new DERTaggedObject(true, 503, AnoAuthRequired);
-            var creationDateTime = new DERTaggedObject(true, 701, AcreationDateTime);
-            var origin = new DERTaggedObject(true, 702, Aorigin);
-            var rootOfTrust = new DERTaggedObject(true, 704, rootOfTrustSeq);
-            var osVersion = new DERTaggedObject(true, 705, AosVersion);
-            var osPatchLevel = new DERTaggedObject(true, 706, AosPatchLevel);
-            var applicationID = new DERTaggedObject(true, 709, AapplicationID);
-            var vendorPatchLevel = new DERTaggedObject(true, 718, AvendorPatchLevel);
-            var bootPatchLevel = new DERTaggedObject(true, 719, AbootPatchlevel);
-
-            ASN1Encodable[] teeEnforcedEncodables;
-
-            // Support device properties attestation
-            if (params.brand != null) {
-                var Abrand = new DEROctetString(params.brand);
-                var Adevice = new DEROctetString(params.device);
-                var Aproduct = new DEROctetString(params.product);
-                var Amanufacturer = new DEROctetString(params.manufacturer);
-                var Amodel = new DEROctetString(params.model);
-                var brand = new DERTaggedObject(true, 710, Abrand);
-                var device = new DERTaggedObject(true, 711, Adevice);
-                var product = new DERTaggedObject(true, 712, Aproduct);
-                var manufacturer = new DERTaggedObject(true, 716, Amanufacturer);
-                var model = new DERTaggedObject(true, 717, Amodel);
-
-                teeEnforcedEncodables = new ASN1Encodable[]{purpose, algorithm, keySize, digest, ecCurve,
-                        noAuthRequired, origin, rootOfTrust, osVersion, osPatchLevel, vendorPatchLevel,
-                        bootPatchLevel, brand, device, product, manufacturer, model};
-            } else {
-                teeEnforcedEncodables = new ASN1Encodable[]{purpose, algorithm, keySize, digest, ecCurve,
-                        noAuthRequired, origin, rootOfTrust, osVersion, osPatchLevel, vendorPatchLevel,
-                        bootPatchLevel};
-            }
-
-            ASN1Encodable[] softwareEnforced = {applicationID, creationDateTime};
-
-            ASN1OctetString keyDescriptionOctetStr = getAsn1OctetString(teeEnforcedEncodables, softwareEnforced, params);
-
-            return new Extension(new ASN1ObjectIdentifier("1.3.6.1.4.1.11129.2.1.17"), false, keyDescriptionOctetStr);
-        } catch (Throwable t) {
-            Log.e(TAG, Log.getStackTraceString(t));
-        }
-        return null;
-    }
-
-    private static int getOsVersion() {
-        String release = Build.VERSION.RELEASE;
-        int major = 0, minor = 0, patch = 0;
-
-        String[] parts = release.split("\\.");
-        if (parts.length > 0) major = Integer.parseInt(parts[0]);
-        if (parts.length > 1) minor = Integer.parseInt(parts[1]);
-        if (parts.length > 2) patch = Integer.parseInt(parts[2]);
-
-        return major * 10000 + minor * 100 + patch;
-    }
-
-    private static int getPatchLevel() {
-        return convertPatchLevel(Build.VERSION.SECURITY_PATCH, false);
-    }
-
-    private static int getPatchLevelLong() {
-        return convertPatchLevel(Build.VERSION.SECURITY_PATCH, true);
-    }
-
-    private static int convertPatchLevel(String patchLevel, boolean longFormat) {
-        try {
-            String[] parts = patchLevel.split("-");
-            int year = Integer.parseInt(parts[0]);
-            int month = Integer.parseInt(parts[1]);
-            if (longFormat) {
-                int day = Integer.parseInt(parts[2]);
-                return year * 10000 + month * 100 + day;
-            } else {
-                return year * 100 + month;
-            }
-        } catch (Exception e) {
-            Log.e(TAG, "Invalid patch level: " + patchLevel, e);
-            return 202404;
-        }
-    }
-
-    private static ASN1OctetString getAsn1OctetString(ASN1Encodable[] teeEnforcedEncodables, ASN1Encodable[] softwareEnforcedEncodables, KeyGenParameters params) throws IOException {
-        ASN1Integer attestationVersion = new ASN1Integer(100);
-        ASN1Enumerated attestationSecurityLevel = new ASN1Enumerated(1);
-        ASN1Integer keymasterVersion = new ASN1Integer(100);
-        ASN1Enumerated keymasterSecurityLevel = new ASN1Enumerated(1);
-        ASN1OctetString attestationChallenge = new DEROctetString(params.attestationChallenge);
-        ASN1OctetString uniqueId = new DEROctetString("".getBytes());
-        ASN1Encodable softwareEnforced = new DERSequence(softwareEnforcedEncodables);
-        ASN1Sequence teeEnforced = new DERSequence(teeEnforcedEncodables);
-
-        ASN1Encodable[] keyDescriptionEncodables = {attestationVersion, attestationSecurityLevel, keymasterVersion,
-                keymasterSecurityLevel, attestationChallenge, uniqueId, softwareEnforced, teeEnforced};
-
-        ASN1Sequence keyDescriptionHackSeq = new DERSequence(keyDescriptionEncodables);
-
-        return new DEROctetString(keyDescriptionHackSeq);
-    }
-
-    private static DEROctetString createApplicationId(int uid) throws Throwable {
-        Context context = ActivityThread.currentApplication();
-        if (context == null) {
-            throw new IllegalStateException("createApplicationId: context not available from ActivityThread!");
-        }
-
-        PackageManager pm = context.getPackageManager();
-        if (pm == null) {
-            throw new IllegalStateException("createApplicationId: PackageManager not found!");
-        }
-
-        String[] packages = pm.getPackagesForUid(uid);
-        if (packages == null || packages.length == 0) {
-            throw new IllegalStateException("No packages found for UID: " + uid);
-        }
-
-        int size = packages.length;
-        ASN1Encodable[] packageInfoAA = new ASN1Encodable[size];
-        Set<Digest> signatures = new HashSet<>();
-        MessageDigest dg = MessageDigest.getInstance("SHA-256");
-
-        for (int i = 0; i < size; i++) {
-            String name = packages[i];
-            PackageInfo info = pm.getPackageInfo(name, PackageManager.GET_SIGNATURES);
-            ASN1Encodable[] arr = new ASN1Encodable[2];
-            arr[ATTESTATION_PACKAGE_INFO_PACKAGE_NAME_INDEX] =
-                    new DEROctetString(name.getBytes(StandardCharsets.UTF_8));
-            arr[ATTESTATION_PACKAGE_INFO_VERSION_INDEX] =
-                    new ASN1Integer(info.getLongVersionCode());
-            packageInfoAA[i] = new DERSequence(arr);
-
-            for (Signature s : info.signatures) {
-                signatures.add(new Digest(dg.digest(s.toByteArray())));
-            }
-        }
-
-        ASN1Encodable[] signaturesAA = new ASN1Encodable[signatures.size()];
-        int i = 0;
-        for (Digest d : signatures) {
-            signaturesAA[i++] = new DEROctetString(d.digest);
-        }
-
-        ASN1Encodable[] applicationIdAA = new ASN1Encodable[2];
-        applicationIdAA[ATTESTATION_APPLICATION_ID_PACKAGE_INFOS_INDEX] =
-                new DERSet(packageInfoAA);
-        applicationIdAA[ATTESTATION_APPLICATION_ID_SIGNATURE_DIGESTS_INDEX] =
-                new DERSet(signaturesAA);
-
-        return new DEROctetString(new DERSequence(applicationIdAA).getEncoded());
-    }
-
-    record Digest(byte[] digest) {
-        @Override
-        public boolean equals(@Nullable Object o) {
-            if (o instanceof Digest d)
-                return Arrays.equals(digest, d.digest);
-            return false;
-        }
-
-        @Override
-        public int hashCode() {
-            return Arrays.hashCode(digest);
-        }
-    }
-
-    private static KeyPair buildECKeyPair(KeyGenParameters params) throws Exception {
-        Security.removeProvider(BouncyCastleProvider.PROVIDER_NAME);
-        Security.addProvider(new BouncyCastleProvider());
-        ECGenParameterSpec spec = new ECGenParameterSpec(params.ecCurveName);
-        KeyPairGenerator kpg = KeyPairGenerator.getInstance("EC", BouncyCastleProvider.PROVIDER_NAME);
-        kpg.initialize(spec);
-        return kpg.generateKeyPair();
-    }
-
-    private static KeyPair buildRSAKeyPair(KeyGenParameters params) throws Exception {
-        Security.removeProvider(BouncyCastleProvider.PROVIDER_NAME);
-        Security.addProvider(new BouncyCastleProvider());
-        RSAKeyGenParameterSpec spec = new RSAKeyGenParameterSpec(
-                params.keySize, params.rsaPublicExponent);
-        KeyPairGenerator kpg = KeyPairGenerator.getInstance("RSA", BouncyCastleProvider.PROVIDER_NAME);
-        kpg.initialize(spec);
-        return kpg.generateKeyPair();
-    }
-
-    private static void dlog(String msg) {
-        if (DEBUG) Log.d(TAG, msg);
-    }
-
-    public static class KeyGenParameters {
-        public int keySize;
-        public int algorithm;
-        public BigInteger certificateSerial;
-        public Date certificateNotBefore;
-        public Date certificateNotAfter;
-        public X500Name certificateSubject;
-
-        public BigInteger rsaPublicExponent;
-        public int ecCurve;
-        public String ecCurveName;
-
-        public List<Integer> purpose = new ArrayList<>();
-        public List<Integer> digest = new ArrayList<>();
-
-        public byte[] attestationChallenge;
-        public byte[] brand;
-        public byte[] device;
-        public byte[] product;
-        public byte[] manufacturer;
-        public byte[] model;
-
-        public int securityLevel;
-
-        public KeyGenParameters(KeyParameter[] params) {
-            for (KeyParameter kp : params) {
-                switch (kp.tag) {
-                    case Tag.KEY_SIZE -> keySize = kp.value.getInteger();
-                    case Tag.ALGORITHM -> algorithm = kp.value.getAlgorithm();
-                    case Tag.CERTIFICATE_SERIAL -> certificateSerial = new BigInteger(kp.value.getBlob());
-                    case Tag.CERTIFICATE_NOT_BEFORE -> certificateNotBefore = new Date(kp.value.getDateTime());
-                    case Tag.CERTIFICATE_NOT_AFTER -> certificateNotAfter = new Date(kp.value.getDateTime());
-                    case Tag.CERTIFICATE_SUBJECT -> certificateSubject =
-                            new X500Name(new X500Principal(kp.value.getBlob()).getName());
-                    case Tag.RSA_PUBLIC_EXPONENT -> rsaPublicExponent = BigInteger.valueOf(kp.value.getLongInteger());
-                    case Tag.EC_CURVE -> {
-                        ecCurve = kp.value.getEcCurve();
-                        ecCurveName = getEcCurveName(ecCurve);
-                    }
-                    case Tag.PURPOSE -> {
-                        purpose.add(kp.value.getKeyPurpose());
-                    }
-                    case Tag.DIGEST -> {
-                        digest.add(kp.value.getDigest());
-                    }
-                    case Tag.ATTESTATION_CHALLENGE -> attestationChallenge = kp.value.getBlob();
-                    case Tag.ATTESTATION_ID_BRAND -> brand = kp.value.getBlob();
-                    case Tag.ATTESTATION_ID_DEVICE -> device = kp.value.getBlob();
-                    case Tag.ATTESTATION_ID_PRODUCT -> product = kp.value.getBlob();
-                    case Tag.ATTESTATION_ID_MANUFACTURER -> manufacturer = kp.value.getBlob();
-                    case Tag.ATTESTATION_ID_MODEL -> model = kp.value.getBlob();
-                    case Tag.HARDWARE_TYPE -> securityLevel = kp.value.getSecurityLevel();
-                }
-            }
-        }
-
-        private static String getEcCurveName(int curve) {
-            String res;
-            switch (curve) {
-                case EcCurve.CURVE_25519 -> res = "CURVE_25519";
-                case EcCurve.P_224 -> res = "secp224r1";
-                case EcCurve.P_256 -> res = "secp256r1";
-                case EcCurve.P_384 -> res = "secp384r1";
-                case EcCurve.P_521 -> res = "secp521r1";
-                default -> throw new IllegalArgumentException("unknown curve");
-            }
-            return res;
-        }
-    }
-}
diff --git a/core/java/com/android/internal/util/custom/KeyboxImitationHooks.java b/core/java/com/android/internal/util/custom/KeyboxImitationHooks.java
deleted file mode 100644
index 135d2e63b269..000000000000
--- a/core/java/com/android/internal/util/custom/KeyboxImitationHooks.java
+++ /dev/null
@@ -1,177 +0,0 @@
-/*
- * SPDX-FileCopyrightText: 2024 Paranoid Android
- * SPDX-FileCopyrightText: 2025 Neoteric OS
- * SPDX-License-Identifier: Apache-2.0
- */
-package com.android.internal.util.custom;
-
-import android.hardware.security.keymint.Algorithm;
-import android.hardware.security.keymint.KeyParameter;
-import android.hardware.security.keymint.KeyParameterValue;
-import android.hardware.security.keymint.Tag;
-import android.os.Binder;
-import android.system.keystore2.Authorization;
-import android.system.keystore2.IKeystoreSecurityLevel;
-import android.system.keystore2.KeyDescriptor;
-import android.system.keystore2.KeyEntryResponse;
-import android.system.keystore2.KeyMetadata;
-import android.util.Log;
-
-import com.android.internal.util.custom.KeyboxChainGenerator.KeyGenParameters;
-
-import java.security.cert.Certificate;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.LinkedList;
-import java.util.List;
-
-/**
- * @hide
- */
-public class KeyboxImitationHooks {
-
-    private static final String TAG = "KeyboxImitationHooks";
-    private static final boolean DEBUG = Log.isLoggable(TAG, Log.DEBUG);
-    private static boolean mSuccess = false;
-
-    public static KeyEntryResponse onGetKeyEntry(KeyDescriptor descriptor) {
-        if (!KeyProviderManager.isKeyboxAvailable()) {
-            return null;
-        }
-
-        if (!mSuccess) {
-            return null;
-        }
-
-        KeyEntryResponse spoofed = KeyboxUtils.retrieve(Binder.getCallingUid(), descriptor.alias);
-        if (spoofed != null) {
-            dlog("Key entry spoofed");
-            return spoofed;
-        }
-
-        return null;
-    }
-
-    public static KeyMetadata generateKey(IKeystoreSecurityLevel level, KeyDescriptor descriptor, Collection<KeyParameter> args) {
-        if (!KeyProviderManager.isKeyboxAvailable()) {
-            return null;
-        }
-
-        KeyGenParameters params = new KeyGenParameters(args.toArray(new KeyParameter[args.size()]));
-
-        if (params.attestationChallenge == null) {
-            return null;
-        }
-
-        if (params.algorithm != Algorithm.EC && params.algorithm != Algorithm.RSA) {
-            Log.w(TAG, "Unsupported algorithm: " + params.algorithm);
-            return null;
-        }
-
-        int uid = Binder.getCallingUid();
-        try {
-            List<Certificate> chain = KeyboxChainGenerator.generateCertChain(uid, descriptor, params);
-            if (chain == null || chain.isEmpty()) {
-                return null;
-            }
-            KeyEntryResponse response = buildResponse(level, chain, params, descriptor);
-            if (response == null) {
-                return null;
-            }
-            KeyboxUtils.append(uid, descriptor.alias, response);
-            mSuccess = true;
-            return response.metadata;
-        } catch (Exception e) {
-            Log.e(TAG, "Failed to generate key", e);
-            return null;
-        }
-    }
-
-    private static KeyEntryResponse buildResponse(
-            IKeystoreSecurityLevel level,
-            List<Certificate> chain,
-            KeyGenParameters params,
-            KeyDescriptor descriptor
-    ) {
-        try {
-            KeyEntryResponse response = new KeyEntryResponse();
-            KeyMetadata metadata = new KeyMetadata();
-            metadata.keySecurityLevel = params.securityLevel;
-
-            KeyboxUtils.putCertificateChain(metadata, chain.toArray(new Certificate[chain.size()]));
-
-            KeyDescriptor d = new KeyDescriptor();
-            d.domain = descriptor.domain;
-            d.nspace = descriptor.nspace;
-            metadata.key = d;
-
-            List<Authorization> authorizations = new ArrayList<>();
-            Authorization a;
-
-            for (Integer i : params.purpose) {
-                a = new Authorization();
-                a.keyParameter = new KeyParameter();
-                a.keyParameter.tag = Tag.PURPOSE;
-                a.keyParameter.value = KeyParameterValue.keyPurpose(i);
-                a.securityLevel = params.securityLevel;
-                authorizations.add(a);
-            }
-
-            for (Integer i : params.digest) {
-                a = new Authorization();
-                a.keyParameter = new KeyParameter();
-                a.keyParameter.tag = Tag.DIGEST;
-                a.keyParameter.value = KeyParameterValue.digest(i);
-                a.securityLevel = params.securityLevel;
-                authorizations.add(a);
-            }
-
-            a = new Authorization();
-            a.keyParameter = new KeyParameter();
-            a.keyParameter.tag = Tag.ALGORITHM;
-            a.keyParameter.value = KeyParameterValue.algorithm(params.algorithm);
-            a.securityLevel = params.securityLevel;
-            authorizations.add(a);
-
-            a = new Authorization();
-            a.keyParameter = new KeyParameter();
-            a.keyParameter.tag = Tag.KEY_SIZE;
-            a.keyParameter.value = KeyParameterValue.integer(params.keySize);
-            a.securityLevel = params.securityLevel;
-            authorizations.add(a);
-
-            a = new Authorization();
-            a.keyParameter = new KeyParameter();
-            a.keyParameter.tag = Tag.EC_CURVE;
-            a.keyParameter.value = KeyParameterValue.ecCurve(params.ecCurve);
-            a.securityLevel = params.securityLevel;
-            authorizations.add(a);
-
-            a = new Authorization();
-            a.keyParameter = new KeyParameter();
-            a.keyParameter.tag = Tag.NO_AUTH_REQUIRED;
-            a.keyParameter.value = KeyParameterValue.boolValue(true); // TODO: copy
-            a.securityLevel = params.securityLevel;
-            authorizations.add(a);
-
-            // TODO: ORIGIN, OS_VERSION, OS_PATCHLEVEL, VENDOR_PATCHLEVEL, BOOT_PATCHLEVEL,
-            // CREATION_DATETIME, USER_ID
-
-            metadata.authorizations = authorizations.toArray(new Authorization[0]);
-            response.metadata = metadata;
-            response.iSecurityLevel = level;
-            return response;
-        } catch (Exception e) {
-            Log.e(TAG, "Failed to build key entry response", e);
-            return null;
-        }
-    }
-
-    public static void setSuccessFlag(boolean flag) {
-        mSuccess = flag;
-    }
-
-    private static void dlog(String msg) {
-        if (DEBUG) Log.d(TAG, msg);
-    }
-}
diff --git a/core/java/com/android/internal/util/custom/KeyboxUtils.java b/core/java/com/android/internal/util/custom/KeyboxUtils.java
deleted file mode 100644
index 178714bf2704..000000000000
--- a/core/java/com/android/internal/util/custom/KeyboxUtils.java
+++ /dev/null
@@ -1,154 +0,0 @@
-/*
- * SPDX-FileCopyrightText: 2025 Neoteric OS
- * SPDX-License-Identifier: Apache-2.0
- */
-package com.android.internal.util.custom;
-
-import android.security.keystore.KeyProperties;
-import android.system.keystore2.KeyEntryResponse;
-import android.system.keystore2.KeyMetadata;
-
-import com.android.internal.org.bouncycastle.asn1.ASN1Sequence;
-import com.android.internal.org.bouncycastle.asn1.ASN1Primitive;
-import com.android.internal.org.bouncycastle.asn1.DERNull;
-import com.android.internal.org.bouncycastle.asn1.pkcs.PKCSObjectIdentifiers;
-import com.android.internal.org.bouncycastle.asn1.pkcs.PrivateKeyInfo;
-import com.android.internal.org.bouncycastle.asn1.pkcs.RSAPrivateKey;
-import com.android.internal.org.bouncycastle.asn1.sec.ECPrivateKey;
-import com.android.internal.org.bouncycastle.asn1.x9.X9ObjectIdentifiers;
-import com.android.internal.org.bouncycastle.asn1.x509.AlgorithmIdentifier;
-import com.android.internal.org.bouncycastle.cert.X509CertificateHolder;
-
-import java.io.ByteArrayInputStream;
-import java.io.ByteArrayOutputStream;
-import java.security.KeyFactory;
-import java.security.PrivateKey;
-import java.security.cert.Certificate;
-import java.security.cert.CertificateFactory;
-import java.security.cert.X509Certificate;
-import java.security.spec.PKCS8EncodedKeySpec;
-import java.security.spec.RSAPrivateCrtKeySpec;
-import java.util.ArrayList;
-import java.util.Base64;
-import java.util.List;
-import java.util.concurrent.ConcurrentHashMap;
-
-/**
- * @hide
- */
-public class KeyboxUtils {
-
-    private static final ConcurrentHashMap<Key, KeyEntryResponse> response = new ConcurrentHashMap<>();
-    public static record Key(int uid, String alias) {}
-
-    public static byte[] decodePemOrBase64(String input) {
-        String base64 = input
-                .replaceAll("-----BEGIN [^-]+-----", "")
-                .replaceAll("-----END [^-]+-----", "")
-                .replaceAll("\\s+", "");
-        return Base64.getDecoder().decode(base64);
-    }
-
-    public static PrivateKey parsePrivateKey(String encodedKey, String algorithm) throws Exception {
-        byte[] keyBytes = decodePemOrBase64(encodedKey);
-        ASN1Primitive primitive = ASN1Primitive.fromByteArray(keyBytes);
-        if ("EC".equalsIgnoreCase(algorithm)) {
-            try {
-                // Try parsing as PKCS#8
-                PrivateKeyInfo info = PrivateKeyInfo.getInstance(primitive);
-                return KeyFactory.getInstance("EC").generatePrivate(new PKCS8EncodedKeySpec(info.getEncoded()));
-            } catch (Exception e) {
-                // Possibly SEC1 / PKCS#1 EC
-                ASN1Sequence seq = ASN1Sequence.getInstance(primitive);
-                ECPrivateKey ecPrivateKey = ECPrivateKey.getInstance(seq);
-                AlgorithmIdentifier algId = new AlgorithmIdentifier(X9ObjectIdentifiers.id_ecPublicKey, ecPrivateKey.getParameters());
-                PrivateKeyInfo privInfo = new PrivateKeyInfo(algId, ecPrivateKey);
-                PKCS8EncodedKeySpec pkcs8Spec = new PKCS8EncodedKeySpec(privInfo.getEncoded());
-                return KeyFactory.getInstance("EC").generatePrivate(pkcs8Spec);
-            }
-        } else if ("RSA".equalsIgnoreCase(algorithm)) {
-            try {
-                // Try parsing as PKCS#8
-                return KeyFactory.getInstance("RSA").generatePrivate(new PKCS8EncodedKeySpec(keyBytes));
-            } catch (Exception e) {
-                // Parse as PKCS#1
-                RSAPrivateKey rsaKey = RSAPrivateKey.getInstance(primitive);
-                AlgorithmIdentifier algId = new AlgorithmIdentifier(PKCSObjectIdentifiers.rsaEncryption, DERNull.INSTANCE);
-                PrivateKeyInfo privInfo = new PrivateKeyInfo(algId, rsaKey);
-                PKCS8EncodedKeySpec pkcs8Spec = new PKCS8EncodedKeySpec(privInfo.getEncoded());
-                return KeyFactory.getInstance("RSA").generatePrivate(pkcs8Spec);
-            }
-        } else {
-            throw new IllegalArgumentException("Unsupported algorithm: " + algorithm);
-        }
-    }
-
-    public static X509Certificate parseCertificate(String encodedCert) throws Exception {
-        byte[] certBytes = decodePemOrBase64(encodedCert);
-        return (X509Certificate) CertificateFactory
-                .getInstance("X.509")
-                .generateCertificate(new ByteArrayInputStream(certBytes));
-    }
-
-    public static List<Certificate> getCertificateChain(String algorithm) throws Exception {
-        IKeyboxProvider provider = KeyProviderManager.getProvider();
-        String[] certChainPem = KeyProperties.KEY_ALGORITHM_EC.equals(algorithm)
-                ? provider.getEcCertificateChain()
-                : provider.getRsaCertificateChain();
-
-        CertificateFactory factory = CertificateFactory.getInstance("X.509");
-        List<Certificate> certs = new ArrayList<>();
-
-        for (String certPem : certChainPem) {
-            certs.add(parseCertificate(certPem));
-        }
-
-        return certs;
-    }
-
-    public static void putCertificateChain(KeyEntryResponse response, Certificate[] chain) throws Exception {
-        putCertificateChain(response.metadata, chain);
-    }
-
-    public static void putCertificateChain(KeyMetadata metadata, Certificate[] chain) throws Exception {
-        metadata.certificate = chain[0].getEncoded();
-        var output = new ByteArrayOutputStream();
-        for (int i = 1; i < chain.length; i++) {
-            output.write(chain[i].getEncoded());
-        }
-        metadata.certificateChain = output.toByteArray();
-    }
-
-    public static X509Certificate getCertificateFromHolder(X509CertificateHolder holder) throws Exception {
-        CertificateFactory certFactory = CertificateFactory.getInstance("X.509");
-        ByteArrayInputStream in = new ByteArrayInputStream(holder.getEncoded());
-        return (X509Certificate) certFactory.generateCertificate(in);
-    }
-
-    public static PrivateKey getPrivateKey(String algorithm) throws Exception {
-        IKeyboxProvider provider = KeyProviderManager.getProvider();
-        String privateKeyEncoded = KeyProperties.KEY_ALGORITHM_EC.equals(algorithm)
-                ? provider.getEcPrivateKey()
-                : provider.getRsaPrivateKey();
-
-        return parsePrivateKey(privateKeyEncoded, algorithm);
-    }
-
-    public static X509CertificateHolder getCertificateHolder(String algorithm) throws Exception {
-        IKeyboxProvider provider = KeyProviderManager.getProvider();
-        String certPem = KeyProperties.KEY_ALGORITHM_EC.equals(algorithm)
-                ? provider.getEcCertificateChain()[0]
-                : provider.getRsaCertificateChain()[0];
-
-        X509Certificate parsedCert = parseCertificate(certPem);
-        return new X509CertificateHolder(parsedCert.getEncoded());
-    }
-
-    public static void append(int uid, String a, KeyEntryResponse c) {
-        response.put(new Key(uid, a), c);
-    }
-
-    public static KeyEntryResponse retrieve(int uid, String a) {
-        return response.get(new Key(uid, a));
-    }
-}
diff --git a/core/res/res/values/custom_config.xml b/core/res/res/values/custom_config.xml
index fd879ef43e5f..a0fe4c4769c9 100644
--- a/core/res/res/values/custom_config.xml
+++ b/core/res/res/values/custom_config.xml
@@ -109,45 +109,6 @@
     <bool name="config_dt2sGestureAvailable">false</bool>
     <bool name="config_dt2sGestureEnabledByDefault">false</bool>
 
-    <!-- Build properties from a GMS certified device against current platform SPL level -->
-    <string-array name="config_certifiedBuildProperties" translatable="false">
-      <!--
-        Each entry must be of the format
-            FIELD:value
-        with the field belonging to android.os.Build or android.os.Build.VERSION class.
-        Example:
-        <item>BRAND:foo</item>
-        <item>DEVICE:bar</item>
-        <item>FINGERPRINT:foo/bar/bar:1.0/lorem/ipsum:dolor/sit-amet</item>
-        <item>VERSION.RELEASE:1.0</item>
-      -->
-    </string-array>
-
-    <!-- Keybox configuration for device certification -->
-    <string-array name="config_certifiedKeybox" translatable="false">
-      <!--
-        Keybox entries for EC and RSA keys and certificates.
-        These should be populated with valid data for device certification.
-        <item>EC.PRIV:</item>
-        <item>EC.CERT_1:</item>
-        <item>EC.CERT_2:</item>
-        <item>EC.CERT_3:</item>
-        <item>RSA.PRIV:</item>
-        <item>RSA.CERT_1:</item>
-        <item>RSA.CERT_2:</item>
-        <item>RSA.CERT_3:</item>
-      -->
-    </string-array>
-
-    <!-- Build fingerprint from stock ROM -->
-    <string name="config_stockFingerprint" translatable="false"></string>
-
-    <!-- Device model (Build.MODEL) to spoof for netflix, if not empty -->
-    <string name="config_netflixSpoofModel" translatable="false"></string>
-
-   <!-- API URL to fetch PIF from -->
-   <string name="config_pifUpdateUrl" translatable="false">https://raw.githubusercontent.com/PixelOS-AOSP/official_devices/refs/heads/sixteen/API/updater/CertifiedProps/certified_build_props.json</string>
-
    <!-- Defines the actions shown in advanced reboot submenu -->
    <string-array name="config_restartActionsList">
        <item>restart</item>
diff --git a/core/res/res/values/custom_symbols.xml b/core/res/res/values/custom_symbols.xml
index 5865b7cbc9d4..91738a8384e3 100644
--- a/core/res/res/values/custom_symbols.xml
+++ b/core/res/res/values/custom_symbols.xml
@@ -101,15 +101,6 @@
      <java-symbol type="string" name="url_copy_success" />
      <java-symbol type="string" name="url_copy_failed" />
 
-    <!-- Prop/KeyboxImitationHooks -->
-    <java-symbol type="array" name="config_certifiedBuildProperties" />
-    <java-symbol type="array" name="config_certifiedKeybox" />
-    <java-symbol type="string" name="config_stockFingerprint" />
-    <java-symbol type="string" name="config_netflixSpoofModel" />
-
-   <!-- API URL to fetch PIF from -->
-   <java-symbol type="string" name="config_pifUpdateUrl" />
-
    <!-- Defines the actions shown in advanced reboot submenu -->
    <java-symbol type="array" name="config_restartActionsList" />
 </resources>
diff --git a/keystore/java/android/security/KeyStore2.java b/keystore/java/android/security/KeyStore2.java
index 77cd77994adb..f5cf571ad955 100644
--- a/keystore/java/android/security/KeyStore2.java
+++ b/keystore/java/android/security/KeyStore2.java
@@ -32,8 +32,6 @@ import android.system.keystore2.KeyEntryResponse;
 import android.system.keystore2.ResponseCode;
 import android.util.Log;
 
-import com.android.internal.util.custom.KeyboxImitationHooks;
-
 import java.util.Calendar;
 
 /**
@@ -285,11 +283,6 @@ public class KeyStore2 {
             throws KeyStoreException {
         StrictMode.noteDiskRead();
 
-        KeyEntryResponse response = KeyboxImitationHooks.onGetKeyEntry(descriptor);
-        if (response != null) {
-            return response;
-        }
-
         return handleRemoteExceptionWithRetry((service) -> service.getKeyEntry(descriptor));
     }
 
diff --git a/keystore/java/android/security/KeyStoreSecurityLevel.java b/keystore/java/android/security/KeyStoreSecurityLevel.java
index da5c0cc04595..6ab148a8b4ea 100644
--- a/keystore/java/android/security/KeyStoreSecurityLevel.java
+++ b/keystore/java/android/security/KeyStoreSecurityLevel.java
@@ -33,8 +33,6 @@ import android.system.keystore2.KeyMetadata;
 import android.system.keystore2.ResponseCode;
 import android.util.Log;
 
-import com.android.internal.util.custom.KeyboxImitationHooks;
-
 import java.util.Calendar;
 import java.util.Collection;
 
@@ -148,15 +146,6 @@ public class KeyStoreSecurityLevel {
             throws KeyStoreException {
         StrictMode.noteDiskWrite();
 
-        KeyboxImitationHooks.setSuccessFlag(false);
-        if (attestationKey == null) {
-            KeyMetadata metadata = KeyboxImitationHooks.generateKey(mSecurityLevel,
-                    descriptor, args);
-            if (metadata != null) {
-                return metadata;
-            }
-        }
-
         return handleExceptions(() -> mSecurityLevel.generateKey(
                 descriptor, attestationKey, args.toArray(new KeyParameter[args.size()]),
                 flags, entropy));
diff --git a/keystore/java/android/security/keystore2/AndroidKeyStoreSpi.java b/keystore/java/android/security/keystore2/AndroidKeyStoreSpi.java
index 81ca30ea8676..e6a63b9c4c17 100644
--- a/keystore/java/android/security/keystore2/AndroidKeyStoreSpi.java
+++ b/keystore/java/android/security/keystore2/AndroidKeyStoreSpi.java
@@ -48,7 +48,6 @@ import android.system.keystore2.ResponseCode;
 import android.util.Log;
 
 import com.android.internal.annotations.VisibleForTesting;
-import com.android.internal.util.PropImitationHooks;
 
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
@@ -179,8 +178,6 @@ public class AndroidKeyStoreSpi extends KeyStoreSpi {
 
     @Override
     public Certificate[] engineGetCertificateChain(String alias) {
-        PropImitationHooks.onEngineGetCertificateChain();
-
         KeyEntryResponse response = getKeyMetadata(alias);
 
         if (response == null || response.metadata.certificate == null) {
diff --git a/packages/SettingsProvider/src/android/provider/settings/validators/SecureSettingsValidators.java b/packages/SettingsProvider/src/android/provider/settings/validators/SecureSettingsValidators.java
index 8c356247faa1..52e3d5100fc2 100644
--- a/packages/SettingsProvider/src/android/provider/settings/validators/SecureSettingsValidators.java
+++ b/packages/SettingsProvider/src/android/provider/settings/validators/SecureSettingsValidators.java
@@ -521,8 +521,5 @@ public class SecureSettingsValidators {
         VALIDATORS.put(Secure.DOZE_FOR_NOTIFICATIONS, BOOLEAN_VALIDATOR);
         VALIDATORS.put(Secure.DOZE_TAP_GESTURE_AMBIENT, BOOLEAN_VALIDATOR);
         VALIDATORS.put(Secure.DOZE_PICK_UP_GESTURE_AMBIENT, BOOLEAN_VALIDATOR);
-        VALIDATORS.put(Secure.KEYBOX_DATA, ANY_STRING_VALIDATOR);
-        VALIDATORS.put(Secure.PIF_DATA, ANY_STRING_VALIDATOR);
-        VALIDATORS.put(Secure.FETCHED_PIF, ANY_STRING_VALIDATOR);
     }
 }
diff --git a/services/core/java/com/android/server/custom/AttestationService.java b/services/core/java/com/android/server/custom/AttestationService.java
deleted file mode 100644
index 6cbd23b1344b..000000000000
--- a/services/core/java/com/android/server/custom/AttestationService.java
+++ /dev/null
@@ -1,149 +0,0 @@
-/*
- * Copyright (C) 2024 The LeafOS Project
- *
- * SPDX-License-Identifier: Apache-2.0
- *
- */
-
-package com.android.server.custom;
-
-import android.content.Context;
-import android.content.pm.PackageManager;
-import android.content.res.Resources;
-import android.net.ConnectivityManager;
-import android.net.Network;
-import android.net.NetworkCapabilities;
-import android.os.Environment;
-import android.os.SystemProperties;
-import android.provider.Settings;
-import android.util.Log;
-
-import com.android.internal.util.custom.CustomUtils;
-import com.android.server.SystemService;
-
-import java.io.BufferedReader;
-import java.io.File;
-import java.io.FileReader;
-import java.io.FileWriter;
-import java.io.IOException;
-import java.io.InputStreamReader;
-import java.net.HttpURLConnection;
-import java.net.URI;
-import java.net.URL;
-import java.util.concurrent.Executors;
-import java.util.concurrent.ScheduledExecutorService;
-import java.util.concurrent.TimeUnit;
-
-public final class AttestationService extends SystemService {
-    private static final String TAG = AttestationService.class.getSimpleName();
-    private static final String API =
-            Resources.getSystem().getString(com.android.internal.R.string.config_pifUpdateUrl);
-
-    private static final String DATA_FILE = "gms_certified_props.json";
-
-    private static final long INITIAL_DELAY = 0;
-    private static final long INTERVAL = 5;
-
-    private static final boolean DEBUG = Log.isLoggable(TAG, Log.DEBUG);
-    private static final Boolean sDisableGmsProps =
-            SystemProperties.getBoolean("persist.sys.pihooks.disable.gms_props", false);
-
-    private final Context mContext;
-    private final ScheduledExecutorService mScheduler;
-
-    public AttestationService(Context context) {
-        super(context);
-        mContext = context;
-        mScheduler = Executors.newSingleThreadScheduledExecutor();
-    }
-
-    @Override
-    public void onStart() {}
-
-    @Override
-    public void onBootPhase(int phase) {
-        if (!sDisableGmsProps && CustomUtils.isPackageInstalled(mContext, "com.google.android.gms")
-                && phase == PHASE_BOOT_COMPLETED) {
-            Log.i(TAG, "Scheduling the service");
-            mScheduler.scheduleAtFixedRate(
-                    new FetchGmsCertifiedProps(), INITIAL_DELAY, INTERVAL, TimeUnit.MINUTES);
-        }
-    }
-
-    private String fetchProps() {
-        try {
-            URL url = new URI(API).toURL();
-            HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
-
-            try {
-                urlConnection.setConnectTimeout(10000);
-                urlConnection.setReadTimeout(10000);
-
-                try (BufferedReader reader = new BufferedReader(
-                             new InputStreamReader(urlConnection.getInputStream()))) {
-                    StringBuilder response = new StringBuilder();
-                    String line;
-
-                    while ((line = reader.readLine()) != null) {
-                        response.append(line);
-                    }
-
-                    return response.toString();
-                }
-            } finally {
-                urlConnection.disconnect();
-            }
-        } catch (Exception e) {
-            Log.e(TAG, "Error making an API request", e);
-            return null;
-        }
-    }
-
-    private boolean isInternetConnected() {
-        ConnectivityManager cm =
-                (ConnectivityManager) mContext.getSystemService(Context.CONNECTIVITY_SERVICE);
-        Network nw = cm.getActiveNetwork();
-        if (nw == null)
-            return false;
-        NetworkCapabilities actNw = cm.getNetworkCapabilities(nw);
-        return actNw != null
-                && (actNw.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)
-                        || actNw.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)
-                        || actNw.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET)
-                        || actNw.hasTransport(NetworkCapabilities.TRANSPORT_BLUETOOTH));
-    }
-
-    private void dlog(String message) {
-        if (DEBUG)
-            Log.d(TAG, message);
-    }
-
-    private class FetchGmsCertifiedProps implements Runnable {
-        @Override
-        public void run() {
-            try {
-                dlog("FetchGmsCertifiedProps started");
-
-                if (!isInternetConnected()) {
-                    Log.e(TAG, "Internet unavailable");
-                    return;
-                }
-
-                String savedProps = Settings.Secure.getString(
-                        mContext.getContentResolver(), Settings.Secure.FETCHED_PIF);
-                String props = fetchProps();
-
-                if (props != null && !savedProps.equals(props)) {
-                    dlog("Found new props");
-                    Settings.Secure.putString(
-                            mContext.getContentResolver(), Settings.Secure.FETCHED_PIF, props);
-                    dlog("FetchGmsCertifiedProps completed");
-                } else {
-                    dlog("No change in props");
-                }
-            } catch (Exception e) {
-                Log.e(TAG, "Error in FetchGmsCertifiedProps", e);
-            }
-        }
-    }
-}
diff --git a/services/core/java/com/android/server/wm/ActivityTaskManagerService.java b/services/core/java/com/android/server/wm/ActivityTaskManagerService.java
index 8e764db1addf..04e01502bc0a 100644
--- a/services/core/java/com/android/server/wm/ActivityTaskManagerService.java
+++ b/services/core/java/com/android/server/wm/ActivityTaskManagerService.java
@@ -264,7 +264,6 @@ import com.android.internal.protolog.ProtoLog;
 import com.android.internal.util.ArrayUtils;
 import com.android.internal.util.FastPrintWriter;
 import com.android.internal.util.FrameworkStatsLog;
-import com.android.internal.util.PropImitationHooks;
 import com.android.internal.util.function.pooled.PooledLambda;
 import com.android.server.LocalManagerRegistry;
 import com.android.server.LocalServices;
@@ -2043,9 +2042,7 @@ public class ActivityTaskManagerService extends IActivityTaskManager.Stub {
 
     @Override
     public RootTaskInfo getFocusedRootTaskInfo() throws RemoteException {
-        if (!PropImitationHooks.shouldBypassTaskPermission(mContext)) {
-            enforceTaskPermission("getFocusedRootTaskInfo()");
-        }
+        enforceTaskPermission("getFocusedRootTaskInfo()");
         final long ident = Binder.clearCallingIdentity();
         try {
             synchronized (mGlobalLock) {
@@ -3229,9 +3226,7 @@ public class ActivityTaskManagerService extends IActivityTaskManager.Stub {
     /** Sets the task stack listener that gets callbacks when a task stack changes. */
     @Override
     public void registerTaskStackListener(ITaskStackListener listener) {
-        if (!PropImitationHooks.shouldBypassTaskPermission(mContext)) {
-            enforceTaskPermission("registerTaskStackListener()");
-        }
+        enforceTaskPermission("registerTaskStackListener()");
         mTaskChangeNotificationController.registerTaskStackListener(listener);
     }
 
diff --git a/services/java/com/android/server/SystemServer.java b/services/java/com/android/server/SystemServer.java
index 9f2907d065e3..f8661bd941e5 100644
--- a/services/java/com/android/server/SystemServer.java
+++ b/services/java/com/android/server/SystemServer.java
@@ -218,7 +218,6 @@ import com.android.server.pdb.PersistentDataBlockService;
 import com.android.server.people.PeopleService;
 import com.android.server.permission.access.AccessCheckingService;
 import com.android.server.pinner.PinnerService;
-import com.android.server.custom.AttestationService;
 import com.android.server.pm.ApexManager;
 import com.android.server.pm.ApexSystemServiceInfo;
 import com.android.server.pm.BackgroundInstallControlService;
@@ -2891,11 +2890,6 @@ public final class SystemServer implements Dumpable {
             t.traceBegin("StartHealthService");
             mSystemServiceManager.startService(HealthInterfaceService.class);
             t.traceEnd();
-
-            // AttestationService
-            t.traceBegin("AttestationService");
-            mSystemServiceManager.startService(AttestationService.class);
-            t.traceEnd();
         }
 
         t.traceBegin("StartMediaProjectionManager");
-- 
2.34.1

